### ISSUE TITLE
Problem when resolving object datakeyref

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2620]
[datakeyrefProblem.zip](https://github.com/dita-ot/dita-ot/files/821184/datakeyrefProblem.zip)

My DITA Map refers to a topic and has a key which points to a movie:

        <topicref href="topics/movies.dita"/>
        <keydef href="resources/Particles.mp4" keys="shining_dots" format="mp4"/>

My topic is in the folder "topics/movies.dita" and has a reference to a video like this:

     <object datakeyref="shining_dots"/>

When producing HTML output the datakeyref is converted to:

    <object data="resources/Particles.mp4"></object>

but it should actually be:

    <object data="../resources/Particles.mp4"></object>

in order to properly point to the video.

**Comment by raducoravu** [comment]
Still happens with DITA OT 3.1.

**Comment by raducoravu** [comment]
👍 Another one of our end users reported this today.

**Comment by robander** [comment]
The problem seems to be for non-DITA keys (not specific to `<object>`). 

I took the original sample above, and added an `xref` reference using the same key:
```
<object datakeyref="shining_dots" outputclass="video"/>
<p>As a test, try xref: <xref keyref="shining_dots">link link</xref></p>
```

The relative path was wrong for both -- the link and the object are both missing `../`.

I also tried using a key value that refers to a DITA topic in the resources directory. (That is invalid on `@datakeyref` but I wanted to compare results.) When the target is DITA, both `@keyref` and `@datakeyref` resolve using the correct relative path to `../resources/ditatest.html` in generated HTML files.

**Comment by robander** [comment]
Guessing (but haven't tested) that the issue is in here ... the keyref parsing code checks to see if the key target `isLocalDita` [here](https://github.com/dita-ot/dita-ot/blob/develop/src/main/java/org/dita/dost/writer/KeyrefPaser.java#L526), and if so, it does a bunch of relative path calculation (along with a lot of other stuff); the mp4 key would fall into the `else` condition just below, which doesn't change any paths even when the file is local:
https://github.com/dita-ot/dita-ot/blob/develop/src/main/java/org/dita/dost/writer/KeyrefPaser.java#L554

The same issue comes up with `scope="peer"` links using valid relative paths -- if I try the same DITA key above but add `scope="peer"`, the eventual link value will not be adjusted, regardless of where the key is used. I think this has been reported in other older issues but they may have been closed by `probot`.

**Comment by robander** [comment]
I don't know which issue is best left open, but I've gone down a rabbit hole now and am finding duplicates that all come down to the same part of keyref resolution:

- The scope=peer one I remembered was indeed closed by probot #1951 
- #2581 is the same issue (keyref to a PDF breaks when key is resolved in another directory)
- #2250 is an image key (which works for images due to special handling of images, but the reported defect is about linking to the image with xref)
